### PR TITLE
FLINK-4287: Add owner label to k8s crs

### PR DESCRIPTION
**Pull request by AhmedSoliman7**
To fix PaaSTA labels violations, the owner label must be added to the k8s CRs. 

The mapping of kinds to owners will be added internally in this [PR](https://github.yelpcorp.com/sysgit/puppet/pull/8482)

### CODE REVIEW

**Review by AshishKhatkar — APPROVED** [review]
looks good to me

File: paasta_tools/setup_kubernetes_cr.py [hunk 1]
```diff
@@ -240,6 +240,13 @@ def get_dashboard_base_url(kind: str, cluster: str) -> Optional[str]:
     return None
 
 
+# FLINK-4287: This should be more generalized later to include owners of other kinds
+def get_cr_owner(kind: str) -> Optional[str]:
+    if kind.lower() == "flink":
```
Comment (gmdfalk) [inline paasta_tools/setup_kubernetes_cr.py:245]: would you mind adding kafkacluster, cassandracluster and nrtsearchservice?

**Review by gmdfalk — CHANGES_REQUESTED** [review]

File: paasta_tools/setup_kubernetes_cr.py (hunk 1 above)
Comment (AhmedSoliman7) [reply paasta_tools/setup_kubernetes_cr.py:245]: Do you have the owner names? I am not sure about the names here.

Comment (gmdfalk) [reply paasta_tools/setup_kubernetes_cr.py:245]: My mapping of ownership is:
```
jenkins (cid)
jolt (cid)
paasta (compute_infra_platform_experience)
paasta-cassandraclusters (database-reliability-engineering-nosql)
paasta-flinks (stream-processing)
paasta-kafkaclusters (kafka-operator)
paasta-monkrelays (streaming-platform)
paasta-nrtsearchservices (ranking_platform)
paasta-scyllaclusters (cid)
paasta-spark (core_ml)
tron (compute_infra_platform_experience)
rest/system_stuff (compute_infra_core)
```

Comment (gmdfalk) [reply paasta_tools/setup_kubernetes_cr.py:245]: so `cassandracluster -> database-reliability-engineering-nosql`, `flink -> stream-processing`, `kafkacluster -> kafka-operator`, , `monkrelay -> streaming-platform` (no CRD yet, i think but cant hurt), `nrtsearchservice -> ranking_platform` and `scyllacluster -> cid`

Comment (AhmedSoliman7) [reply paasta_tools/setup_kubernetes_cr.py]: Done

File: paasta_tools/setup_kubernetes_cr.py [hunk 2]
```diff
@@ -275,7 +276,9 @@ def format_custom_resource(
     url = get_dashboard_base_url(kind, cluster)
     if url:
         resource["metadata"]["annotations"][paasta_prefixed("dashboard_base_url")] = url
-
+    owner = CRDS_KIND_TO_OWNER[kind.lower()]
```
Comment (marcos-sb) [inline paasta_tools/setup_kubernetes_cr.py:279]: I think we'll get a `KeyError` if `kind.lower()` is not present in the dict. Using `get()` should fix that.

```suggestion
    owner = CRDS_KIND_TO_OWNER.get(kind.lower())
```

**Review by marcos-sb — APPROVED** [review]
nice! 🚀

File: paasta_tools/setup_kubernetes_cr.py (hunk 2 above)
Comment (AhmedSoliman7) [reply paasta_tools/setup_kubernetes_cr.py]: Great spot!

**Comment by AhmedSoliman7** [comment]
@gmdfalk Not sure why it is failing tbh..

File: paasta_tools/setup_kubernetes_cr.py (hunk 1 above)
Comment (AhmedSoliman7) [reply paasta_tools/setup_kubernetes_cr.py]: Are we sure kafkacluster owner would be `kafka-operator`?

**Review by tub — APPROVED** [review]
lgtm as long as the tests pass :)

**Review by marcos-sb — APPROVED** [review]
lgtm

File: paasta_tools/setup_kubernetes_cr.py [hunk 3]
```diff
@@ -275,7 +281,9 @@ def format_custom_resource(
     url = get_dashboard_base_url(kind, cluster)
     if url:
         resource["metadata"]["annotations"][paasta_prefixed("dashboard_base_url")] = url
-
+    owner = get_cr_owner(kind.lower())
```
Comment (nemacysts) [inline paasta_tools/setup_kubernetes_cr.py:284]: nit: we're already lowercasing in `get_cr_owner()`, should we drop the lowercasing here?

File: paasta_tools/utils.py [hunk 4]
```diff
@@ -2152,6 +2152,9 @@ def get_cluster(self) -> str:
     def get_dashboard_links(self) -> Mapping[str, Mapping[str, str]]:
         return self.config_dict["dashboard_links"]
 
+    def get_cr_owners(self) -> Mapping[str, str]:
```
Comment (nemacysts) [inline paasta_tools/utils.py:2155]: nit:
```suggestion
    def get_cr_owners(self) -> Dict[str, str]:
```

we're unlikely to return any other mapping collection here, so we might as well explicitly type this as returning a dict

**Review by nemacysts — APPROVED** [review]
I think this will cause k8s to bounce anything that needs a label applied, we'll probably want to apply the internal PR workload-by-workload

File: paasta_tools/utils.py [hunk 5]
```diff
@@ -2152,6 +2152,9 @@ def get_cluster(self) -> str:
     def get_dashboard_links(self) -> Mapping[str, Mapping[str, str]]:
         return self.config_dict["dashboard_links"]
 
+    def get_cr_owners(self) -> Dict[str, str]:
+        return self.config_dict["cr_owners"]
```
Comment (nemacysts) [inline paasta_tools/utils.py:2156]: mypy says `paasta_tools/utils.py:2156: error: TypedDict "SystemPaastaConfigDict" has no key 'cr_owners'` - we'll need to add an entry for `cr_owners` to https://github.com/Yelp/paasta/blob/ee0d2bd94d8707420dc7ac18a17e78c6f8a729a8/paasta_tools/utils.py#L1885

**Review by tub — APPROVED** [review]

File: unique-run [hunk 6]
```diff
@@ -0,0 +1,18 @@
+apiVersion: v1
```
Comment (nemacysts) [inline unique-run:1]: i'm assuming this wasn't meant to be included?

Comment (AhmedSoliman7) [reply unique-run:1]: Yes, removing.

**Review by nemacysts — APPROVED** [review]

**Pull request merged**